Tests/Shared/Core/UnitTestBase.swift

```diff
@@ -165,13 +165,7 @@ open class UnitTestBase<Component>: XCTestCase {
     }
 
     public func runOnBackground(_ block: @escaping () -> Void) {
-        CioThreadUtil().runBackground {
-            block()
-        }
-    }
-
-    public func runOnMain(_ block: @escaping () -> Void) {
-        CioThreadUtil().runMain {
+        Task {
```

Comment:
This is the bug fix. Switching from GCD to swift concurrency to see if that increases the reliability of the test suite.

Reply:
How are we calculating the reliability? because the CI run before this PR also passed?

Reply:
While working on this ticket, I have run the test suite on the CI a lot.

Before this PR, I was noticing the test suite that would call `runOnMain` or `runOnBackground` would fail 75% of the time. The fact that the test suite has executed 4 times thus far on this PR and none of the tests have failed thus far, that shows increased reliability to me.